packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -502,10 +507,46 @@ You've created a plugin project that doesn't yet support any platforms.
 ''');
 }
 
-void _printAddPlatformMessage(String pluginPath) {
+void _printPluginAddPlatformMessage(String pluginPath) {
   globals.printStatus('''
 To add platforms, run `flutter create -t plugin --platforms <platforms> .` under $pluginPath.
 For more information, see https://flutter.dev/go/plugin-platforms.
 
 ''');
 }
+
+// shows warning if user requested a platform and if it's not enabled
+void _printWarningToEnablePlatforms(
+  List<String> requestedPlatforms,
+  [bool isApp = false]
+) {
+  final List<String> _showWarning = <String>[
+  if (requestedPlatforms.contains('android') && !featureFlags.isAndroidEnabled)
+    'android',
+  if (requestedPlatforms.contains('ios') && !featureFlags.isIOSEnabled)
+    'ios',
+  if (requestedPlatforms.contains('web') && !featureFlags.isWebEnabled)
+    'web',
+  if (requestedPlatforms.contains('macos') && !featureFlags.isMacOSEnabled)
+    'macos',
+  if (requestedPlatforms.contains('windows') && !featureFlags.isWindowsEnabled)
+    'windows',
+  if (requestedPlatforms.contains('linux') && !featureFlags.isLinuxEnabled)
+    'linux',
+  ];
+
+  if (_showWarning.isEmpty){
+    return;
+  }
+
+  final String _disabledPlatforms = _showWarning.join(', ');
+
+  globals.printStatus('! $_disabledPlatforms are not enabled; enable the platforms and then add them.',
```

Comment:
This isn't entirely correct - the user could be on a channel that does not support the platform.

Reply:
I'm not sure how I could get which channels are supported. Can please guide me?

In [this line](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/features.dart#L104) I can see the implementation of getting the channels. But couldn't find the any other way.